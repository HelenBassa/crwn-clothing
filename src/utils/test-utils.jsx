import { render } from '@testing-library/react';
import { BrowserRouter as Router } from "react-router-dom";
import { Provider } from "react-redux";
import { store } from "../store/store";


afterAll(() => jest.clearAllMocks());

const AllTheProviders = ({ children }) => {
  return (
    <Provider store={store}>
      <Router>
        {children}
      </Router>
    </Provider>
  );
};

const customRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';
export * from '@testing-library/jest-dom';
export * from '@testing-library/user-event';

export { customRender as render };