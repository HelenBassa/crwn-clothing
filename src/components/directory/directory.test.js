import { render } from "../../utils/test-utils";
import Directory from "./directory.component";

test("should render Directory component", () => {
  const { container } = render(<Directory />);
  expect(container).toMatchSnapshot();
});
