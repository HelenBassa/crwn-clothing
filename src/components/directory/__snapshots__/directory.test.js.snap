// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Directory component 1`] = `
<div>
  <div
    class="sc-hLBbgP bpLFCn"
  >
    <div
      class="sc-dkrFOg kupPSc"
    >
      <div
        class="sc-bcXHqe mwYYS"
      />
      <div
        class="sc-gswNZR cGctEJ"
      >
        <h2>
          hats
        </h2>
        <p>
          Shop now
        </p>
      </div>
    </div>
    <div
      class="sc-dkrFOg kupPSc"
    >
      <div
        class="sc-bcXHqe hnerAS"
      />
      <div
        class="sc-gswNZR cGctEJ"
      >
        <h2>
          jackets
        </h2>
        <p>
          Shop now
        </p>
      </div>
    </div>
    <div
      class="sc-dkrFOg kupPSc"
    >
      <div
        class="sc-bcXHqe iWkgMR"
      />
      <div
        class="sc-gswNZR cGctEJ"
      >
        <h2>
          sneakers
        </h2>
        <p>
          Shop now
        </p>
      </div>
    </div>
    <div
      class="sc-dkrFOg kupPSc"
    >
      <div
        class="sc-bcXHqe gASsa-d"
      />
      <div
        class="sc-gswNZR cGctEJ"
      >
        <h2>
          womens
        </h2>
        <p>
          Shop now
        </p>
      </div>
    </div>
    <div
      class="sc-dkrFOg kupPSc"
    >
      <div
        class="sc-bcXHqe dRsiuS"
      />
      <div
        class="sc-gswNZR cGctEJ"
      >
        <h2>
          mens
        </h2>
        <p>
          Shop now
        </p>
      </div>
    </div>
  </div>
</div>
`;
